python/private/interpreter_hub.bzl

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 The Bazel Authors. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"Repo rule used by bzlmod extension to create a repo that has a map of Python interpreters and there labels"
+
+load("//python:versions.bzl", "WINDOWS_NAME")
+load("//python/private:toolchains_repo.bzl", "get_host_os_arch", "get_host_platform")
+
+_build_file_for_hub_template = """
+INTERPRETER_LABELS = {{
+{lines}
+}}
+"""
+
+_line_for_hub_template = """\
+    "{name}": Label("@{name}_{platform}//:{path}"),
```

Comment:
Hm, if the key is a simple string, I wonder what happens if different modules register the same name. Imma try this and see what happens.

Reply:
Hah, welp, this uncovered some deeper problems, but not related to a string key being used here, and they're pre-existing to this change.

The basic issue is, because all the repos are created in the rules_python module, all the names passed into the rules_python extensions have to be globally unique. e.g., if two submodules both try to do `python.toolchain(name="X")` (same for `interpreter.install`, `pip.parse`, etc), then the name arg has to be globally unique. This makes sense considering the semantics we understand about bzlmod. I'm pretty sure we can fix this, but it's something for another change.

Using a string key has a couple minor issues: a user can use a name someone else registered, and it doesn't handle repo renaming. But I don't think those are particularly big deals.